server/controllers/list.js

```diff
@@ -40,9 +40,16 @@ const createList = (req, resp) => {
     Cohort.findOne({ _id: cohortId })
       .then(cohort => {
         const { memberIds } = cohort;
-        list.viewersIds = memberIds;
 
-        return list.save();
+        if (checkIfArrayContainsUserId(memberIds, userId)) {
+          list.viewersIds = memberIds;
+
+          return list.save();
+        }
+
+        throw new BadRequestException(
```

Comment:
You don't catch this specific error message, you prevent this edge case, the code is stopped here, but in catch statement, you should write something like this:
```
.catch(err => {
      if (err instanceof BadRequestException) {
        const { status, message } = err;

        return resp.status(status).send({ message });
      }

      resp.status(400).send({
        message: 'List not saved. Please try again.'
      });
    });
```